### PR TITLE
Metrics for errors

### PR DESCRIPTION
**Pull request by benrudolph**
Adding some more metrics. I'm envisioning a crash report dashboard @wpride @snopoke

### CODE REVIEW

File: src/main/java/application/AbstractBaseController.java [hunk 1]
```diff
@@ -289,6 +298,15 @@ public ExceptionResponseBean handleError(FormplayerHttpRequest req, Exception ex
         return new ExceptionResponseBean(exception.getMessage(), req.getRequestURL().toString());
     }
 
+    private void incrementDatadogCounter(String metric, FormplayerHttpRequest req) {
+        datadogStatsDClient.increment(
+                metric,
+                "domain:" + req.getDomain(),
+                "user:" + req.getCouchUser().getUsername(),
```
Comment (benrudolph) [inline src/main/java/application/AbstractBaseController.java]: assumption here is that if we ever have a `FormplayerHttpRequest` then we have a user. let me know if you think that's a bad assumption

Comment (snopoke) [inline src/main/java/application/AbstractBaseController.java]: I don't think form validation requests will have a user

Comment (wpride) [inline src/main/java/application/AbstractBaseController.java]: What about serverup?

Comment (benrudolph) [inline src/main/java/application/AbstractBaseController.java]: presumably serverup will never go through error handling, but maybe we should guard against that?

Comment (benrudolph) [inline src/main/java/application/AbstractBaseController.java]: gonna do it for due diligence

**Pull request merged**